vaadin-renderer-flow-parent/vaadin-renderer-flow/src/main/java/com/vaadin/flow/data/renderer/ComponentRenderer.java

```diff
@@ -140,33 +142,70 @@ public ComponentRenderer(
         this.componentUpdateFunction = componentUpdateFunction;
     }
 
-    /**
-     * Default constructor, that can be used by subclasses which supports
-     * different ways of creating components, other than those defined in the
-     * other constructors.
-     */
-    protected ComponentRenderer() {
+    @Override
+    protected String getTemplateExpression() {
+        var appId = UI.getCurrent() != null
+                ? UI.getCurrent().getInternals().getAppId()
+                : "";
+
+        return String.format(
+                "<%s appid=\"%s\" .nodeid=\"${item.nodeid}\"></%s>",
+                componentRendererTag, appId, componentRendererTag);
+    }
+
+    Element getOwner() {
+        return owner;
     }
 
     @Override
-    public Rendering<SOURCE> render(Element container,
-            DataKeyMapper<SOURCE> keyMapper, Element contentTemplate) {
+    public Rendering<SOURCE> render(Element owner,
+            DataKeyMapper<SOURCE> keyMapper, String rendererName) {
+        this.owner = owner;
+        this.container = new Element("div");
+        owner.appendVirtualChild(container);
+        var rendering = super.render(owner, keyMapper, rendererName);
 
-        ComponentRendering rendering = new ComponentRendering(
-                keyMapper == null ? null : keyMapper::key);
-        rendering.setTemplateElement(contentTemplate);
-        /*
-         * setupTemplateWhenAttached does some setup that will be needed by
-         * generateData. To ensure the setup has completed before it is needed,
-         * we forego the general convention of using beforeClientResponse to
-         * guard the action against duplicate invocation. This is not a big
-         * problem in this case since setupTemplateWhenAttached only sets
-         * properties but doesn't execute any JS.
-         */
-        container.getNode().runWhenAttached(ui -> setupTemplateWhenAttached(ui,
-                container, rendering, keyMapper));
+        return configureRendering(rendering, keyMapper);
+    }
 
-        return rendering;
+    /**
+     * Configures the {@code Rendering} instance provided by {@link LitRenderer}
```

Comment:
Nit: Could clarify that this configures the rendering to create / update components for individual items. 

There is also some potential for confusion in `BasicRenderer` which overrides this to not create components. Maybe the method name could reflect the purpose better, but I don't have any good suggestions.

Reply:
Updated the documentation. Also made configureRendering in BasicRenderer API package-private.